tests/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/Properties/KeyValuePairListEncodingTests.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Microsoft.VisualStudio.ProjectSystem.Debug;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties;
+
+public class KeyValuePairListEncodingTests
+{
+    [Theory]
+    [InlineData("key1=value1;key2=value2", new[] { "key1", "value1", "key2", "value2" })]
+    [InlineData("key1=value1;;key2=value2", new[] { "key1", "value1", "key2", "value2" })]
+    [InlineData("key1=value1;;;key2=value2", new[] { "key1", "value1", "key2", "value2" })]
+    [InlineData("key1=value1;key2=value2;key3=value3", new[] { "key1", "value1", "key2", "value2", "key3", "value3" })]
+    [InlineData("key1;key2=value2", new[] { "key1", "", "key2", "value2" })]
+    [InlineData("key1;key2;key3=value3", new[] { "key1", "", "key2", "", "key3", "value3" })]
+    [InlineData("key1;;;key3;;", new[] { "key1", "", "key3", "" })]
```

Comment:
Would be good to cover some more inputs:

- `""`
- `" "`
- `"="`
- `";"`

If there are invalid inputs, a test that ensures that `Parse` throws would be good. For example, `"=="`, or `null`.

Reply:
I added test cases on those inputs except null, as the method accepts a non-nullable input string. But I wouldn't necessarily agree that there are invalid inputs possible here, since `==` should parse to an empty, value `=`

Reply:
Thanks for adding more tests.

> since == should parse to an empty key, value =

I'm less confident about that than you. I feel like `==` should be an invalid and ambiguous input that throws. If values are expected to contain delimiters, then they should be escaped.

Reply:
> I feel like == should be an invalid and ambiguous input that throws.

I'm not entirely in agreement; the difference between = and the other delimiters is that `=` has no meaning other than as part of a value after already encountering an `=`

Reply:
What happens if you enter strings in the UI containing `=` and `,`?

Reply:
Good points. I switched back to escaping the =